test/test_function_schema.py

```diff
@@ -112,6 +112,32 @@ def test_string_optional_parameter_default_value(self):
         schema_b = parse_schema(str(schema_a))
         self.assertEquals(schema_a, schema_b)
 
+    def test_forward_compatible_arguments_without_out(self):
+        old_schema = parse_schema('any(Tensor self, int a, int b=1) -> Tensor')
+        new_schema = parse_schema('any(Tensor self, int a, int b=1, int c=1) -> Tensor')
+        self.assertTrue(new_schema.is_forward_compatible_with(old_schema))
+        new_schema = parse_schema('any(Tensor self, int a, int b=1, int[2] c=1) -> Tensor')
+        self.assertFalse(new_schema.is_forward_compatible_with(old_schema))
```

Comment:
Looks like `isForwardCompatibleWith`'s PyBind definition doesn't allow returning a string of `why_not`. Could we add it and verify match failures are due to the right reasons?

Reply:
That's a good idea. Will do